cmd/root/root.go

```diff
@@ -126,5 +135,106 @@ func Execute(ctx context.Context, cmd *cobra.Command) error {
 		}
 	}
 
+	end := time.Now()
+	exitCode := 0
+	if err != nil {
+		exitCode = 1
+	}
+
+	uploadTelemetry(cmd.Context(), commandString(cmd), start, end, exitCode)
 	return err
 }
+
+// We want child telemetry processes to inherit environment variables like $HOME or $HTTPS_PROXY
+// because they influence auth resolution.
+func inheritEnvVars() []string {
+	base := os.Environ()
+	out := []string{}
+	authEnvVars := auth.EnvVars()
+
+	// Remove any existing auth environment variables. This is done because
+	// the CLI offers multiple modalities of configuring authentication like
+	// `--profile` or `DATABRICKS_CONFIG_PROFILE` or `profile: <profile>` in the
+	// bundle config file.
+	//
+	// Each of these modalities have different priorities and thus we don't want
+	// any auth configuration to piggyback into the child process environment.
+	//
+	// This is a precaution to avoid conflicting auth configurations being passed
+	// to the child telemetry process.
+	for _, v := range base {
+		k, _, found := strings.Cut(v, "=")
+		if !found {
+			continue
+		}
+		if slices.Contains(authEnvVars, k) {
+			continue
+		}
+		out = append(out, v)
+	}
+
+	return out
+}
+
+func uploadTelemetry(ctx context.Context, cmdStr string, start, end time.Time, exitCode int) {
+	// Nothing to upload.
+	if !telemetry.HasLogs(ctx) {
+		return
+	}
+
+	// Telemetry is disabled. We don't upload logs.
+	if _, ok := os.LookupEnv(telemetry.DisableEnvVar); ok {
+		return
+	}
+
+	telemetry.SetExecutionContext(ctx, protos.ExecutionContext{
+		CmdExecID:       cmdExecId,
+		Version:         build.GetInfo().Version,
+		Command:         cmdStr,
+		OperatingSystem: runtime.GOOS,
+		DbrVersion:      env.Get(ctx, dbr.EnvVarName),
+		ExecutionTimeMs: end.Sub(start).Milliseconds(),
+		ExitCode:        int64(exitCode),
+	})
+
+	logs := telemetry.GetLogs(ctx)
+
+	in := telemetry.UploadConfig{
+		Logs: logs,
+	}
+
+	// Compute environment variables with the appropriate auth configuration.
+	env := inheritEnvVars()
+	for k, v := range auth.Env(ConfigUsed(ctx)) {
+		env = append(env, fmt.Sprintf("%s=%s", k, v))
+	}
+
+	d := daemon.Daemon{
+		Args:        []string{"telemetry", "upload"},
+		Env:         env,
+		PidFilePath: os.Getenv(telemetry.PidFileEnvVar),
+		LogFile:     os.Getenv(telemetry.UploadLogsFileEnvVar),
+	}
+
+	err := d.Start()
+	if err != nil {
+		log.Debugf(ctx, "failed to start telemetry worker: %s", err)
```

Comment:
I'd raise this at least to Infof, maybe even to Warnf

Reply:
done